src/core/thread/meshcop_dataset_manager.cpp

```diff
@@ -920,6 +936,39 @@ ThreadError PendingDataset::Set(const Timestamp &aTimestamp, const Message &aMes
 {
     ThreadError error = kThreadError_None;
     uint8_t flags;
+    Tlv tlv;
+    uint16_t offset = aOffset;
+    ActiveTimestampTlv activeTimestamp;
+    NetworkMasterKeyTlv masterKey;
+    const Timestamp *localActiveTimestamp = mNetif.GetActiveDataset().GetNetwork().GetTimestamp();
+
+    // check whether or not to roll back the old Active Timestamp
+    while (offset < aMessage.GetLength())
```

Comment:
We should not need to remember if the Pending Dataset is rolling back the Active Timestamp.  Instead, we should just update `PendingDataset::HandleTimer()` to always set both the `mNetwork` and `mLocal` versions of the Active Dataset.  We could do this by adding a parameter to `ActiveDataset::Set()` that indicates whether (i) a Delay Timer or (ii) receiving a message triggered the call to `ActiveDataset::Set()`.  Thoughts?